Code Samples/Fib/main.cpp

```diff
@@ -1,58 +1,86 @@
 #include <iostream>
-#include <unistd.h>
-#include <sys/types.h>
-#include <pthread.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <time.h>
-#include <string.h>
+#include <thread>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+#include <cstring>
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <unistd.h>  
 
 #include "thread.h"
 
 #define THREAD_COUNT 10
 
-static char block[] = "--block";
-int test = 0;
+static constexpr char block[] = "--block";
+static constexpr char crash[] = "--crash";
+static constexpr char test_flag[] = "--test";
+std::atomic<int> test_count{0};  // Thread-safe counter
+volatile std::sig_atomic_t g_signal_status = 0;
+
+void signal_handler(int signal) {
+    g_signal_status = signal;
+}
 
 int main(int argc, char **argv)
 {
-    srand(time(NULL));
+    std::signal(SIGINT, signal_handler);
 
-    static char pidText[] = "PID: ";
-    std::string helpText = "Attach a debugger and execute 'set foo=0' to continue";
-    char helloText[] = "Hello World!";
+    std::srand(static_cast<unsigned>(std::time(nullptr)));
 
-    std::cout << helloText << std::endl;
+    std::cout << "Hello World!" << std::endl;
 
-    pthread_t threads[THREAD_COUNT];
+    if (argc == 2) {
+        if (std::strcmp(block, argv[1]) == 0) {
+            std::cout << "Attach a debugger and set foo=0 to continue" << std::endl;
+            std::cout << "Process ID: " << getpid() << std::endl;
 
-    if (argc == 2 && !strcmp(block, argv[1]))
-    {
-        std::cout << helpText << std::endl;
-        volatile int foo = 1;
-        while (foo)
-            ;
+            volatile int foo = 1;
+            while (foo && g_signal_status == 0) {
+                std::this_thread::sleep_for(std::chrono::seconds(1));
+                std::cout << "Waiting... (press Ctrl-C to quit)" << std::endl;
+            }
+            return 0;
+        }
+        else if (std::strcmp(crash, argv[1]) == 0) {
+            std::cout << "Triggering intentional crash..." << std::endl;
+            volatile int foo = 0;
```

Comment:
I believe the only reason these `foo` and `bar` variables needed to be volatile was to mitigate the behavior of the debugger here.  Since you're using `std::atomic` now (increasing the minimum required language standard version), I'd suggest also converting these uses of 'volatile' to `std::atomic` as well.